contracts/extensions/CoinMachine.sol

```diff
@@ -190,17 +199,23 @@ contract CoinMachine is ColonyExtension {
       "coin-machine-unauthorised"
     );
 
-    uint256 tokenBalance = getTokenBalance();
-    uint256 numTokens = min(min(_numTokens, maxPerPeriod - activeSold), tokenBalance);
+    uint256 maxPurchase = getMaxPurchase(msg.sender);
+    uint256 numTokens = min(maxPurchase, _numTokens);
```

Comment:
Or maybe return - such a transaction could be sent in good faith, and a return would stop the events being emitted and use as little gas as possible.

Reply:
Yes, I agree with the latter thinking. I generally lean to not erroring for zero values, since transactions might be sent in good faith (i.e. the period just sold out).